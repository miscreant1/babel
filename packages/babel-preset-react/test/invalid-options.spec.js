import { loadOptions } from "@babel/core";
import presetReact from "..";

function loadPresetWithOptions(options: any) {
  loadOptions({
    filename: "/fake/test.jsx",
    presets: [[presetReact, options]],
  });
}
describe("preset-react", () => {
  it("should throw when top level options are invalid", () => {
    expect(() => {
      loadPresetWithOptions({ useBuiltins: true });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when boolean options are not provided with boolean values", () => {
    expect(() => {
      loadPresetWithOptions({ useBuiltIns: "true" });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when string options are not provided with string values", () => {
    expect(() => {
      loadPresetWithOptions({ pragma: 0 });
    }).toThrowErrorMatchingSnapshot();
  });
});
