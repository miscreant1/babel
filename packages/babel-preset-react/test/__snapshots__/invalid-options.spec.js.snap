// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-react should throw when boolean options are not provided with boolean values 1`] = `"[BABEL] /fake/test.jsx: @babel/preset-react: 'useBuiltIns' option must be a boolean. (While processing: \\"base$0\\")"`;

exports[`preset-react should throw when string options are not provided with string values 1`] = `"[BABEL] /fake/test.jsx: @babel/preset-react: 'pragma' option must be a string. (While processing: \\"base$0\\")"`;

exports[`preset-react should throw when top level options are invalid 1`] = `
"[BABEL] /fake/test.jsx: @babel/preset-react: 'useBuiltins' is not a valid top-level option.
Maybe you meant to use 'useBuiltIns'? (While processing: \\"base$0\\")"
`;
